### ISSUE TITLE
`namespaces` rule doesn't respect scopes

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#1167]
repro:
```js
import * as wizards from 'jsx/redux/MaxBox/wizards'

// `wizards` defined in local scope
function ComponentX({ wizards, submitWizard }) {
  return (
    <div>
      { shouldShowWizard && !isEmpty(wizards)
        // unable to validate computed reference report here at `wizards[0]`
        ? <WizardZero {...wizards[0]} onSubmit={submitWizard} /> 
        : <WhateverElse />
      }
    </div>
  )
}
```

pretty corner-y case but just making a note
